### PR TITLE
Refactor Pollbook backend to seperate Networking and Internal usages / apis

### PR DESCRIPTION
**Pull request by carolinemodic**
## Overview
https://github.com/votingworks/vxsuite/issues/6152
Refactors the VxPollBook backend so that two seperate processes are run, serving express apps on two different ports. One for local intra-node calls and use cases and one for peer inter-node calls and use cases. 

While the Store class could have been kept as 1 class where the Local and Peer apps are each holding their own instance of it, it seemed like a clearer reminder to the developer that you can not share anything in memory between the processes to seperate them. However various pieces of common functionality are still required as implemented in the parent class. 

Quick overview: 
- index.ts starts both a peerServer (peer_server.ts) and localServer (server.ts) 
- Each server builds its respective express app with a grout api defined (app.ts and peer_app.ts), and the app has a connection to the relevant Store subclass (local_store.ts or peer_store.ts each inheriting from store.ts) 
- The stores talk to the same sqlite database path to share information. 

Note on the tests:
- app_multi_node is configured multiple full apps to do all networking logic, this is good for testing the networking connection logic but is somewhat brittle and hard to manage for testing details of event syncing
- store_multi_node handles when to call to sync events manually rather then testing the polling logic, this makes it easier to manipulate the logic into specific scenarios and is the easier/better place to test that specific events are synced in the right manner

## Testing Plan
Scale Testing: https://docs.google.com/document/d/1oMgYbPZSssFHlLTtT70QtBfVxnxOT0e3rF6LSFzUM2M/edit?tab=t.0#heading=h.j3f9q5wmlqm 
Did fairly extensive manual testing as a part of scale testing
Expanded upon automated tests to cover more edge cases around voter name/address change and registrations events
Expanded upon automated tests to cover basic connection flow for the networking logic 

## Intended Follow-Ups
https://github.com/votingworks/vxsuite/issues/6293 - I saw this occur once in scale testing, not entirely sure the best way to broadly address it, it might require deeper rethink/refactor of db queries and transaction usage which is a bit scattered in implementation right now 
https://github.com/votingworks/vxsuite/issues/6294 - Performance optimization

### CODE REVIEW

File: apps/pollbook/backend/schema.sql [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+    updated_first_name TEXT,
+    updated_last_name TEXT,
     voter_data TEXT not null
 );
 
+CREATE TABLE machines (
+    machine_id TEXT PRIMARY KEY,
+    status TEXT not null,
+    last_updated INTEGER NOT NULL, -- last time the machine was updated
+    last_seen INTEGER NOT NULL -- last time the machine was seen
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/pollbook/backend/schema.sql:14]: confirmed that last_updated is not used right now so going to remove.

File: apps/pollbook/backend/src/app.ts [hunk 2]
```diff
@@ -299,10 +294,13 @@ function buildApi({ context, logger }: BuildAppParams) {
...
 
     getCheckInCounts(): { thisMachine: number; allMachines: number } {
-      return {
+      console.time('getCheckInCounts');
+      const result = {
         thisMachine: store.getCheckInCount(machineId),
         allMachines: store.getCheckInCount(),
-      };
+      } as const;
+      console.timeEnd('getCheckInCounts');
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/pollbook/backend/src/app.ts]: I added general perf timings to grouts built in logging with debug so removing this individual uses.

File: apps/pollbook/backend/src/peer_store.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+export class PeerStore extends Store {
+  private readonly connectedPollbooks: Record<string, PollbookService> = {};
+  private isOnline: boolean = false;
+
+  constructor(client: DbClient, machineId: string) {
+    super(client, machineId);
+
+    // Load connected pollbooks from the database
+    // TODO-CARO-IMPLEMENT should this be reset on reboot?
+    this.client.run(`DELETE FROM machines`);
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/pollbook/backend/src/peer_store.ts:25]: discussed sync

File: apps/pollbook/backend/scripts/simulate_check_ins.ts [hunk 4]
```diff
@@ -61,15 +91,59 @@ async function checkInAllVotersOnCurrentMachine(
     );
 
     let processed = 0;
+    const durations = {
+      searchByInitials: [] as number[],
+      searchByFullName: [] as number[],
+      checkIn: [] as number[],
+    };
+
+    console.time('100processed');
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/pollbook/backend/scripts/simulate_check_ins.ts:100]: this is in the simulation script which is only used for testing so I think it useful to just leave for now

**Review by kshen0 — APPROVED** [review]
